scripts/mbedtls_framework/code_wrapper/psa_wrapper.py

```diff
@@ -0,0 +1,257 @@
+        'psa_pake_set_password_key' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_peer' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_role' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_set_user' : 'defined(PSA_WANT_ALG_SOME_PAKE)',
+        'psa_pake_setup' : 'defined(PSA_WANT_ALG_SOME_PAKE)'
+    }
+}
+
+class PSAWrapper(c_wrapper_generator.Base):
+    """Generate a C source file containing wrapper functions for PSA Crypto API calls."""
+
+    _WRAPPER_NAME_PREFIX = 'mbedtls_test_wrap_'
+    _WRAPPER_NAME_SUFFIX = ''
+
+    _PSA_WRAPPER_INCLUDES = ['<psa/crypto.h>']
+
+    def __init__(self,
+                 output_h_f: str,
+                 output_c_f: str,
+                 in_headers:  Collection[str] = DEFAULTS["input_headers"],
+                 config: Dict[str, Any]= DEFAULTS) -> None:
+
+        super().__init__()
+        self._FUNCTION_GUARDS = super()._FUNCTION_GUARDS.copy()
+        self.in_headers = in_headers
+        self.out_c_f = output_c_f
+        self.out_h_f = output_h_f
+
+        self.mbedtls_root = build_tree.guess_mbedtls_root()
+        self.read_config(config)
+
+        if in_headers:
+            self.read_headers(in_headers)
+
+    def read_config(self, cfg: Dict[str, Any])-> None:
+        """Configure instance's parameters based on a module specific default config """
+
+        self._CPP_GUARDS = PSAWrapper.parse_def_guards(cfg["define_guards"])
+        self._SKIP_FUNCTIONS = cfg["skip_list"]
+        self._FUNCTION_GUARDS.update(cfg["function_guards"]) # type: ignore[arg-type]
+        self._NOT_IMPLEMENTED = cfg["not_implemented"]
+
+    def read_headers(self, headers: Collection[str]) -> None:
+        """ Reads functions from source header files into a Dict[str, FunctionInfo] """
+
+        for header_name in headers:
+            header_path = self.rel_path(header_name)
+            c_parsing_helper.read_function_declarations(self.functions, header_path)
+
+    def rel_path(self, filename: str, path_list: List[str] = ['include', 'psa']) -> str:
+        """ Return the estimated path in relationship to the.
+            mbedtls_root. The method allows overriding the targetted sub-directory.
+            Currently the default is set to mbedtls_root/include/psa """
+
+        # Temporary, while Mbed TLS does not just rely on the TF-PSA-Crypto
+        # build system to build its crypto library. When it does, the first
+        # case can just be removed.
+        if os.path.isdir(os.path.join(self.mbedtls_root, 'tf-psa-crypto')):
+            path_list = ['tf-psa-crypto' ] + path_list
+            return os.path.join(self.mbedtls_root, *path_list, filename)
+
+        return os.path.join(self.mbedtls_root, *path_list, filename)
+
+    # Utility Methods
+    @staticmethod
+    def parse_def_guards(def_list: Collection[str])-> str:
+        """ Parse an input list of format ["HASH_DEFINE", "!HASH_DEFINE2" ] and generate a
```

Comment:
Please don't use the uncommon nickname “hash define”, especially in code where “hash” normally refers to a cryptographic hash algorithm. The official and common name is “[[C] preprocessor] macro”. We sometimes use the somewhat common nickname “[[C] preprocessor] symbol”.